include/git2/diff.h

```diff
@@ -130,7 +130,9 @@ typedef enum {
 	GIT_DIFF_INCLUDE_CASECHANGE = (1u << 11),
 
 	/** If the pathspec is set in the diff options, this flags means to
-	 *  apply it as an exact match instead of as an fnmatch pattern.
```

Comment:
Huh?  What's breaking here?  Can you give an example of something that matches differently?

The change you're commenting on is to better explain the *current* behavior, not document a change.

The goal in this PR is to give a speed up *without* a change in behavior.